server/neptune/gateway/api/request/external/models.go

```diff
@@ -0,0 +1,27 @@
+package external
+
+import validation "github.com/go-ozzo/ozzo-validation"
+
+type Repo struct {
+	Owner string
+	Name  string
+}
+
+func (r Repo) Validate() error {
+	return validation.ValidateStruct(&r,
+		validation.Field(&r.Owner, validation.Required),
+		validation.Field(&r.Name, validation.Required),
+	)
+}
+
+type DeployRequest struct {
+	Roots []string
```

Comment:
can we validate these roots exist in repo?

Reply:
not really, without doing the full clone and parse? 

We do this in RootDeployer though.